src/main/java/com/example/kaizenchat/controller/GroupChatController.java

```diff
@@ -161,4 +159,24 @@ public ResponseEntity<?> getAllChats() {
         }
     }
 
+    @PostMapping("/messages")
+    public ResponseEntity<Map<String, Object>> getLastMessages(@Valid @RequestBody LastMessagesRequest request)
+            throws ChatNotFoundException, UserNotFoundException {
+        var userDetails = (UserDetailsImpl) SecurityContextHolder.getContext()
+                .getAuthentication()
+                .getPrincipal();
+        Long userId = userDetails.getId();
+        log.info("GroupChatController ->  getLastMessages(): user-id={}", userId);
+
+        if(chatService.isMemberInGroupChat(
+                userService.findUserById(userId), chatService.findChatById(request.getChatId(), GROUP))){
+            List<MessageEntity> messages;
+            if(request.getTime() == null)
+                messages = messageService.getLastMessages(request.getChatId(), GET_MESSAGES_LIMIT);
+            else
+                messages = messageService.getLastMessages(request.getChatId(), request.getTime(), GET_MESSAGES_LIMIT);
+            return ResponseEntity.ok().body(Map.of("messages", messages));
+        }else
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(Map.of("message", "not member of chat"));
```

Comment:
[LINE:180] this can be replaced with throwing a UserNotFoundInChatException with same message